pyomo/dataportal/tests/test_dataportal.py

```diff
@@ -791,6 +781,35 @@ def create_options(self, name):
     def create_write_options(self, name):
         return {'filename':os.path.abspath(currdir+os.sep+name+self.suffix), 'sort':True}
 
+    def compare_data(self, name, file_suffix):
+        if file_suffix == '.json':
+            with open(currdir+name+file_suffix, 'r') as out, \
+                open(currdir+name+'.baseline'+file_suffix, 'r') as txt:
+                self.assertStructuredAlmostEqual(json.load(txt), json.load(out),
+                                                 allow_second_superset=True,
+                                                 abstol=0)
+        elif file_suffix == '.yaml':
+            with open(currdir+name+file_suffix, 'r') as out, \
+                open(currdir+name+'.baseline'+file_suffix, 'r') as txt:
+                self.assertStructuredAlmostEqual(yaml.full_load(txt),
+                                                 yaml.full_load(out),
+                                                 allow_second_superset=True,
+                                                 abstol=0)
+        else:
+            try:
+                with open(currdir+name+file_suffix, 'r') as f1, \
+                open(currdir+name+'.baseline'+file_suffix, 'r') as f2:
+                    f1_contents = list(filter(None, f1.read().split()))
+                    f2_contents = list(filter(None, f2.read().split()))
+                    for item1, item2 in zip(f1_contents, f2_contents):
```

Comment:
This should probably use `itertools.zip_longest` so that differing length files cause errors.